module/actor/actor-pc.js

```diff
@@ -71,6 +71,44 @@ export class ArM5ePCActor extends Actor {
     this.data.data.bonuses.traits = { soak: 0 };
   }
 
+  /** @override */
+  prepareEmbeddedDocuments() {
+    if (this.data.type == "laboratory") {
+      this._prepareLaboratoryEmbeddedDocuments(this.data)
+    }
+
+    super.prepareEmbeddedDocuments();
+  }
+
+  _prepareLaboratoryEmbeddedDocuments(labData) {
+    var baseSafetyEffect = labData.effects.find(e => e.data.flags.arm5e.baseSafetyEffect);
+    if (!baseSafetyEffect) {
+      this.createEmbeddedDocuments("ActiveEffect", [
+        {
+          label: game.i18n.localize("arm5e.sheet.baseSafety"),
+          icon: "icons/svg/aura.svg",
+          origin: labData.uuid,
+          tint: "#000000",
+          changes: [
+            {
+              label: "arm5e.sheet.safety",
+              key: "data.safety.bonus",
+              mode: CONST.ACTIVE_EFFECT_MODES.ADD,
+              value: 0
+            }
+          ],
+          flags: {
+            arm5e: {
+              baseSafetyEffect: true,
```

Comment:
I do not really see the issue with having a non-array as flag? This does not seem to cause any problem on my side, besides the `option` array missing.
Having it on a change does allow the specific change to update to be identified, rather than relying on the effect not being modified. But I think that the effect being read-only would work just as well, without the complexity of identifying a specific change to update

Reply:
You changed my mind, beside the option part, it is fine. I am a little paranoid about data corruption. :-P